### PR TITLE
Update Akka.Net project to the latest version

### PR DESCRIPTION
**Pull request by dmitry-golovinov-sonarsource**

### CODE REVIEW

File: analyzers/its/SonarAnalyzer.Testing.ImportBefore.targets [hunk 1]
```diff
@@ -55,8 +55,10 @@
 
     <PropertyGroup Condition="$(SonarAnalyzerRegressionTestRunning)">
       <!-- Set the path to the SARIF report -->
-      <ErrorLog>$(SonarProjectOutputPath)-$(ProjectGuid).json</ErrorLog>
-      <RazorCompilationErrorLog>$(SonarProjectOutputPath)-$(ProjectGuid).Views.json</RazorCompilationErrorLog>
+      <ErrorLog Condition="'$(TargetFramework)' != ''">$(SonarProjectOutputPath)-$(ProjectGuid)-$(TargetFramework).json</ErrorLog>
```
Comment (andrei-epure-sonarsource) [inline analyzers/its/SonarAnalyzer.Testing.ImportBefore.targets:58]: there seems to be an alignment mismatch between the various xml tags (or mix of tab and space?)

Comment (dmitry-golovinov-sonarsource) [reply analyzers/its/SonarAnalyzer.Testing.ImportBefore.targets:58]: Indeed, I should not blindly trust Notepad++ :)

File: analyzers/its/sources/akka.net/src/benchmark/PersistenceBenchmark/PersistenceBenchmark.csproj [hunk 2]
```diff
@@ -1,94 +1,28 @@
...
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
-    <PlatformTarget>AnyCPU</PlatformTarget>
-    <DebugSymbols>true</DebugSymbols>
-    <DebugType>full</DebugType>
-    <Optimize>false</Optimize>
-    <OutputPath>bin\Debug\</OutputPath>
-    <DefineConstants>DEBUG;TRACE</DefineConstants>
-    <ErrorReport>prompt</ErrorReport>
-    <WarningLevel>4</WarningLevel>
-    <CodeAnalysisRuleSet>..\..\..\..\..\ValidationRuleset.ruleset</CodeAnalysisRuleSet>
```
Comment (andrei-epure-sonarsource) [inline analyzers/its/sources/akka.net/src/benchmark/PersistenceBenchmark/PersistenceBenchmark.csproj]: please add this to all CSPROJ files in the solution (both new and old)

I explained here https://github.com/SonarSource/sonar-dotnet/pull/4531#discussion_r648443651 what this is used for

File: analyzers/its/sources/akka.net/src/contrib/cluster/Akka.Cluster.Metrics.Tests/Akka.Cluster.Metrics.Tests.csproj [hunk 3]
```diff
@@ -0,0 +1,31 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+    <Import Project="..\..\..\common.props" />
+    <Import Project="..\..\..\xunitSettings.props" />
+
+    <PropertyGroup>
```
Comment (andrei-epure-sonarsource) [inline analyzers/its/sources/akka.net/src/contrib/cluster/Akka.Cluster.Metrics.Tests/Akka.Cluster.Metrics.Tests.csproj]: test projects need to be manually marked as such : `<SonarQubeTestProject>true</SonarQubeTestProject>`
(and the comment above `<!-- We need to manually mark this project as TEST. See analyzers\its\SonarAnalyzer.Testing.ImportBefore.targets for details.-->`)

**Review by andrei-epure-sonarsource — CHANGES_REQUESTED** [review]

**Comment by andrei-epure-sonarsource** [comment]
these changes should be done in a separate commit

**Review by andrei-epure-sonarsource — COMMENTED** [review]
could you have a look at the wierd EOL character that you've added?
image

File: analyzers/its/sources/akka.net/src/examples/Stocks/SymbolLookup/SymbolLookup.csproj [hunk 4]
```diff
@@ -1,135 +1,135 @@
...
+    <ProjectGuid>{3007A692-A050-4F12-9187-4AE6C6106A66}</ProjectGuid>
+    <OutputType>WinExe</OutputType>
+    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <RootNamespace>SymbolLookup</RootNamespace>
+    <AssemblyName>SymbolLookup</AssemblyName>
+    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
+    <FileAlignment>512</FileAlignment>
+    <SolutionDir Condition="$(SolutionDir) == '' Or $(SolutionDir) == '*Undefined*'">..\..\..\</SolutionDir>
+    <RestorePackages>true</RestorePackages>
    <CodeAnalysisRuleSet>..\..\..\..\..\..\ValidationRuleset.ruleset</CodeAnalysisRuleSet>
```
Comment (andrei-epure-sonarsource) [inline analyzers/its/sources/akka.net/src/examples/Stocks/SymbolLookup/SymbolLookup.csproj]: also each xml element should be on its own line

File: analyzers/its/sources/akka.net/src/contrib/cluster/Akka.Cluster.Metrics.Tests.MultiNode/Akka.Cluster.Metrics.Tests.MultiNode.csproj [hunk 5]
```diff
@@ -1,30 +1,33 @@
...
-        <Service Include="{82a7f48d-3b50-4b1e-b82e-3ada8210c358}" />
-    </ItemGroup>
-
-    <PropertyGroup Condition=" '$(Configuration)' == 'Release' ">
-        <DefineConstants>$(DefineConstants);RELEASE</DefineConstants>
-    </PropertyGroup>
+<Project Sdk="Microsoft.NET.Sdk">
+    <Import Project="..\..\..\common.props" />
+
  <PropertyGroup>
```
Comment (andrei-epure-sonarsource) [inline analyzers/its/sources/akka.net/src/contrib/cluster/Akka.Cluster.Metrics.Tests.MultiNode/Akka.Cluster.Metrics.Tests.MultiNode.csproj]: weird characters here as well in commit a7d687df90

**Review by andrei-epure-sonarsource — CHANGES_REQUESTED** [review]
as discussed offline, it's better to separate 1e165ebd8fb5889a89d20ce5d556db1f7f4818dc and related expected issues changes on a separate PR

(and include also the SonarAnalyzer.Testing.ImportBefore.targets there

**Comment by andrei-epure-sonarsource** [comment]
I make the same comment here as I did for AutoMapper (see [comment](https://github.com/SonarSource/sonar-dotnet/pull/4531#pullrequestreview-679813542)) - is it possible to `git mv` the JSON files which lost their Project GUID after the update?

Because for example `Akka.Cluster.Sharding-{A05C31E8-0246-46A1-B3BC-4D6FE7A9AA49}-S103.json` seems to have been deleted but in fact it has been moved to `Akka.Cluster.Sharding--netstandard2.0-S103.json`

To make it easier:
- sources\akka.net\src\contrib\cluster: Akka.Cluster.Sharding-{A05C31E8-0246-46A1-B3BC-4D6FE7A9AA49}-*.json moved to Akka.Cluster.Sharding--netstandard2.0-*.json
- sources\akka.net\src\contrib\cluster: Akka.Cluster.Tools-{5CF8A8BE-B634-473F-BB01-EBA878746BD4}-*.json moved to Akka.Cluster.Tools--netstandard2.0-*.json
- sources\akka.net\src\contrib\dependencyInjection: Akka.DI.Core-{FDF09D18-B68E-4B95-B1F6-B89D9C6C3AE9}-*.json moved to Akka.DI.Core--netstandard2.0-*.json
- sources\akka.net\src\contrib\persistence: Akka.Persistence.Sql.Common-{3B9E6211-9488-4DB5-B714-24248693B38F}-*.json to Akka.Persistence.Sql.Common--netstandard2.0-*.json
- sources\akka.net\src\contrib\persistence: Akka.Persistence.Sqlite-{453EFD22-7C53-4887-9DBF-FCFC9172E909}-*.json moved to `netstandard2.0` one

sources\akka.net\src\core:
- Akka.Cluster-{6AB00F61-269A-4501-B06A-026707F000A7}-*.json moved to Akka.Cluster--netstandard2.0-*.json
- Akka.MultiNodeTestRunner.Shared-{964F0EC5-FBE6-47C5-8AE6-145114D5DB8C}-*.json moved to Akka.MultiNodeTestRunner.Shared--netstandard2.0-*.json
- Akka.NodeTestRunner-{28520F30-2868-4BD3-9CAE-AC27226C24E3}-*.json moved to Akka.NodeTestRunner--net471-*.json
-  Akka.Persistence-{FCA84DEA-C118-424B-9EB8-34375DFEF18A}-S4784.json moved to `netstandard2.0` one
-  Akka.Remote.Tests.Performance-{453DD4B1-64B8-4B50-BFDA-3489E32D30D8}-*.json moved to `netcoreapp3.1` one
-  Akka.Remote.Tests-{95E921AB-1F5A-4AF9-B7FE-284E317A97F4}-S2699.json moved to net471
-  Akka.Remote-{EA4FF8FD-7C53-49C8-B9AA-02E458B3E6A7}-S927.json moved to netstandard2.0
-  Akka.TestKit-{0D3CBAD0-BBDB-43E5-AFC4-ED1D3ECDC224}-S927.json moved to netstandard2.0
-  /Akka.Tests.Performance-{B17DEDCB-D2AA-472D-82A0-D242B711F488}-S4462.json moved to `netcoreapp3.1`
-  Akka.Tests-{3F786C7D-70E3-4836-8B33-EC9A5AF72330}-S3415.json moved to net471
- Akka-{5DEDDF90-37F0-48D3-A0B0-A5CBD8A7E377}-S927.json moved to netstandard2.0

sources\akka.net\src\examples
- ChatClient-{69AED1DD-02E1-423F-950C-DE6010DFA346}-S105.json moved to `netcoreapp3.1`
- ChatServer-{7FD886CC-216B-4324-9AB2-241494598B06}-S121.json moved to `netcoreapp3.1`
- ClusterSharding.Node-{30A0B17D-B913-49AB-8CBB-8DF383AB7A52}-S4487.json moved to  `netcoreapp3.1`
- Samples.Cluster.Transformation-{E5EB3DF5-D017-4B50-B5AA-2B0440DB773D}-S4142.json moved to net471
- TcpEchoService.Server-{825196A4-4B08-401F-8994-E2DB7C77A8B7}-S4055.json moved to `netcoreapp3.1`

sources\akka.net\src\benchmark\
- PersistenceBenchmark-{39E6F51F-FA1E-4C62-B8F8-19065DE6D55D}-S4462.json moved to net471
- PingPong-{956F7D26-4505-4A26-86D0-73135BD35A93}-S4823.json  moved to net471

If you already have a script to do it, it would be nice to have a clean history.
If it's too difficult, then don't spend time on it and we can merge this as is.

**Review by andrei-epure-sonarsource — APPROVED** [review]
LGTM.

let's avoid spending time on making the git history perfect for these test resources - in practice we won't use the history of the expdected issues in investigation

**Pull request merged**